packages/rum/src/domain/record/observers.spec.ts

```diff
@@ -154,3 +155,150 @@ describe('initFrustrationObserver', () => {
     expect(frustrationsCallbackSpy).not.toHaveBeenCalled()
   })
 })
+
+describe('initStyleSheetObserver', () => {
+  let stopStyleSheetObserver: () => void
+  let styleSheetCallbackSpy: jasmine.Spy<StyleSheetRuleCallback>
+  let styleElement: HTMLStyleElement
+  let styleSheet: CSSStyleSheet
+  const styleRule = '.selector-1 { color: #fff }'
+
+  beforeEach(() => {
+    if (isIE()) {
+      pending('IE not supported')
+    }
+    styleSheetCallbackSpy = jasmine.createSpy()
+    styleElement = document.createElement('style')
+    document.head.appendChild(styleElement)
+    styleSheet = styleElement.sheet!
+
+    serializeDocument(document, NodePrivacyLevel.ALLOW, {
+      status: SerializationContextStatus.INITIAL_FULL_SNAPSHOT,
+      elementsScrollPositions: createElementsScrollPositions(),
+    })
+  })
+
+  afterEach(() => {
+    stopStyleSheetObserver()
+    styleElement.remove()
+  })
+
+  describe('observing high level css stylesheet', () => {
+    describe('when inserting rules into stylesheet', () => {
+      it('should capture CSSStyleRule insertion when no index is provided', () => {
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        styleSheet.insertRule(styleRule)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.removes).toBeUndefined()
+        expect(styleSheetRule.adds?.length).toEqual(1)
+        expect(styleSheetRule.adds?.[0]?.index).toEqual(undefined)
+      })
+
+      it('should capture CSSStyleRule insertion when index is provided', () => {
+        // Given
+        const index = 0
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        styleSheet.insertRule(styleRule, index)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.removes).toBeUndefined()
+        expect(styleSheetRule.adds?.length).toEqual(1)
+        expect(styleSheetRule.adds?.[0]?.index).toEqual(index)
+      })
+    })
+
+    describe('when removing rules from stylesheet', () => {
+      it('should capture CSSStyleRule removal with the correct index', () => {
+        styleSheet.insertRule(styleRule)
+        // Given
+        const index = 0
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        styleSheet.deleteRule(index)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.adds).toBeUndefined()
+        expect(styleSheetRule.removes?.length).toEqual(1)
+        expect(styleSheetRule.removes?.[0]).toEqual({ index })
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Not super fan of adding `given` / `when` / `then` comments, IMO you can separate the setup / action / expectation phases of the test with spacing alone but not a big deal.
However, I am a bit bothered by the instructions that are before the `given` comment, what is the intent there?

Reply:
I test adding and removing CSS Rules at a given index. The steps before `// Given` is a setup of the style sheet.

Reply:
I saw that you removed the comments but don't hesitate if you want to continue the discussion on that.